pkg/provisioner/ironic/ironic.go

```diff
@@ -1219,6 +1220,33 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+
+	p.log.Info("requesting ports to delete")
+
+	err = ports.List(p.client, ports.ListOpts{NodeUUID: p.status.ID}).EachPage(func(page pagination.Page) (provisioner.Result, error) {
+		port_list, err := ports.ExtractPorts(page)
+		if err != nil {
+			return result, errors.Wrap(err, "failed to extract port list")
+		}
+		for i := 0; i < len(port_list) ; i++ {
```

Comment:
It'd be more [idiomatic to use `range`](https://golang.org/doc/effective_go.html#for):

```go
for idx, port := range portList
```